frontend/src/components/Board/Board.js

```diff
@@ -1,13 +1,34 @@
 import React from "react";
 import _ from "lodash";
 import PropTypes from "prop-types";
+import LongPress from "react-long";
 import Field from "./Field/Field";
 import BoardModel from "../../models/BoardModel";
 import "./Board.scss";
+import CircularMenu from "../CircularMenu/CircularMenu";
 
 export default class Board extends React.Component {
   state = {
-    boardModel: new BoardModel(this.props.fields)
+    boardModel: new BoardModel(this.props.fields),
+    suggestions: null
+  };
+
+  getPosition = element => {
+    const rect = element.getBoundingClientRect();
+    return { x: rect.left, y: rect.top };
+  };
+
+  hideSuggestions = () => {
+    this.setState({ suggestions: null });
+  };
+
+  displaySuggestions = (row, column) => {
+    const coords = this.getPosition(
+      document.getElementById(`${row}x${column}`)
```

Comment:
This logic is brittle. A better approach would be to add `onMouseMove` prop to the main board element which updates the position constantly and then you can add `onMouseDown` and `onMouseUp` events to show/hide the menu at the exact position of cursor which was set by the `onMouseMove` handler. This would also show the menu perfectly around your cursor instead of at set positions in elements (not sure if that's the case but that is my deduction from the code).

Reply:
Or, to keep using the long press component, you can just add the `onMouseMove` handlers to have more accurate coordinates instead of getting the hovered element's bounding box.

Reply:
Won't there be a problem with the center of the circular menu? For example when user will click in the corner of the field, the x and y coordinates will be different than when he click in the middle, so the circle will be moved, and will not be simetric.

Reply:
I'm not sure I understand what you are saying. Let's discuss this topic on our next meeting, will be much easier. Perhaps I misunderstood the code.